arcaflow_plugin_stressng/stressng_schema.py

```diff
@@ -25,176 +181,375 @@ class CommonStressorParams:
+        if self.matrix_yx is True:
+            result += "matrix-yx\n"
         return result
 
 
 @dataclass
 class MqStressorParams(CommonStressorParams):
-    mq: int = field(
-        metadata={
-            "name": "MQ count",
-            "description": (
-                "Number of MQ stressors to be run "
-                "(0 means 1 stressor per CPU)"
-            ),
-        }
-    )
+    mq_ops: typing.Annotated[
+        typing.Optional[int],
+        schema.id("mq-ops"),
+        schema.name("MQ Operations"),
+        schema.description(
+            "Stop after N bogo POSIX message send operations completed"
+        ),
+    ] = None
+
+    mq_size: typing.Annotated[
+        typing.Optional[int],
+        schema.id("mq-size"),
+        schema.name("MQ Size"),
+        schema.description(
+            "Specify size of POSIX message queue; the default size is "
+            "10 messages and most Linux systems this is the maximum allowed "
+            "size for normal users"
+        ),
+    ] = None
 
     def to_jobfile(self) -> str:
-        mq = "mq {}\n".format(self.mq)
-        result = mq
+        result = f"mq {self.workers}\n"
+        if self.mq_ops is not None:
+            result += f"mq-ops {self.mq_ops}\n"
+        if self.mq_size is not None:
+            result += f"mq-size {self.mq_size}\n"
         return result
 
 
+@dataclass
+class HDDOptsParams:
+    hdd_opts: typing.Annotated[
+        typing.Optional[HddOpts],
+        schema.name("HDD Options"),
+        schema.description("HDD stress test options"),
+    ] = None
+
+
 @dataclass
 class HDDStressorParams(CommonStressorParams):
-    hdd: int = field(
-        metadata={
-            "name": "HDD workers",
-            "description": (
-                "start N workers continually writing, "
-                "reading and removing temporary files"
-            ),
-        }
-    )
-
-    hdd_bytes: str = field(
-        metadata={
-            "name": "Bytes per worker",
-            "description": (
-                "write  N  bytes for each hdd process, the default is 1 GB. "
-                "One can specify the size in units of Bytes, KBytes, "
-                "MBytes and GBytes using the suffix b, k, m or g."
-            ),
-        }
-    )
-
-    hdd_write_size: str = field(
-        metadata={
-            "name": "Write Size",
-            "description": (
-                "specify size of each write "
-                "in bytes. Size can be from 1 byte to 4MB"
-                "One can specify the size in units of Bytes, KBytes, "
-                "MBytes using the suffix b, k, m"
-            ),
-        }
-    )
+    hdd_bytes: typing.Annotated[
+        typing.Optional[str],
+        schema.id("hdd-bytes"),
+        schema.name("Bytes Per Worker"),
+        schema.description("Write N bytes for each hdd process, the default is 1 GB"),
+    ] = None
+
+    hdd_opts: typing.Annotated[
+        typing.Optional[typing.List[HDDOptsParams]],
```

Comment:
Is this really different than (most of?) the others? Generally we have "this *one* choice, or all available sequentially", but here we get a list, and no mention of what happens if not specified (though it's still `Optional`).

Reply:
Yes, this one is different because the input will accept a list of explicit parameters. The other enums only allow a single selection. And it is accurate that the input of this parameter is `Optional`.